feat: markdownlint: Add a links checker and fix some links

xlai89 opened the pull request:
Add a links checker using [lychee](https://github.com/lycheeverse/lychee/tree/master) into test-markdownlint.sh

Some links are fixed as well.

Fixes #440.

Main reason to choose lychee over [muffet](https://github.com/raviqqe/muffet): 
easy to use. lychee can deal with files, path wildcards and urls, while muffet expects urls as input.

xlai89 commented:
Hi @purpleidea ,
should the following deprecated links just be removed? Didn't manage to update them.

```bash
[docs/on-the-web.md]:
✗ [404] https://www.youtube.com/watch?v=jB992Zb3nH0&html5=1 | Failed: Network error: Not Found
✗ [404] https://www.youtube.com/watch?v=MmpwOQAb_SE&html5=1 | Failed: Network error: Not Found
✗ [404] https://vimeo.com/191493409 | Failed: Network error: Not Found
✗ [ERR] https://roidelapluie.be/blog/2017/02/14/mgmt-augeas/ | Failed: Network error: error sending request for url (https://roidelapluie.be/blog/2017/02/14/mgmt-augeas/)
✗ [ERR] https://annex.debconf.org//debconf-share/debconf16/slides/15-next-generation-config-mgmt.pdf | Failed: Network error: error sending request for url (https://annex.debconf.org//debconf-share/debconf16/slides/15-next-generation-config-mgmt.pdf)
```

purpleidea commented on test/test-markdownlint.sh:
```diff
@@ -16,7 +16,9 @@ if [ -z "$MDL" ]; then
-STYLE=$($mktemp)
+# create a tmp file in current directory e.g. "./tmp.rMKAQ63Cm6"
+# so that mdl command can locate the file (must contain a '/' or end in '.rb')
+STYLE=$($mktemp -p .)
```
Why is this change needed?

xlai89 replied:
After updating mdl to 0.13.0 locally, the test passes as well in my dev env. This change is not necessary and has been removed from the patch.

purpleidea reviewed (commented):
Awesome! One question, and also what do you think we need to do about the timeouts and other issues in the logs? I'll look at the broken links not included in your patch.

Also your commit message should be:

`test: Add a links checker and fix some links`

To pass...

purpleidea commented:
I am just uploading some fixed links, and I will submit a patch to git master with the fixes. After that is done, in theory rebasing your patch should be enough. Thanks again!

purpleidea commented:
Should be ready for rebase.

xlai89 commented:
Thanks for the hint!
Rebase is done. Also corrected the commit message and removed the unnecessary changes.
Can you be more specific about "the timeouts and other issues in the logs" mentioned in an earlier comment?

purpleidea commented:
>  Rebase is done.

Awesome running tests now.

> Also corrected the commit message and removed the unnecessary changes.

Sweet.

> Can you be more specific about "the timeouts and other issues in the logs" mentioned in an earlier comment?

Yes, sorry about my lack of clarity. In the previous build test: https://github.com/purpleidea/mgmt/actions/runs/8958628695/job/24604796138#step:6:178

I see some network timeout issues. Is that a concern? Let's see what this run does. Here's a screenshot:

![image](https://github.com/purpleidea/mgmt/assets/135091/71bb7516-8dfe-409f-ab76-57163698178d)

It occured to me that GH actions might actually rate limit things. If so we might have to disable some stuff. We'll see.

purpleidea commented:
It seems it might be following a non-link but pretending it's a link. Is there a way to disable that?
(And the formatting needs fixing to wrap at 80.)

xlai89 commented:
Sorry for the late reply.

> It seems it might be following a non-link but pretending it's a link. Is there a way to disable that?

Links can be excluded by add them into [`.lycheeignore`](https://github.com/purpleidea/mgmt/pull/758/files#diff-fafe8d160920b6bd97b3c5235607969496ac0550eb4eeb9833393711fc210f18), which contains 2 links for the moment.

> (And the formatting needs fixing to wrap at 80.)

If you mean output of the lychee command, I could change the format with the option `--format` but couldn't limit the length of errors. When it comes to the length of my codes, I somehow didn't find the spot exceeding 80.

purpleidea commented on .lycheeignore:
```diff
@@ -0,0 +1,3 @@
+# list URLs that should be excluded for lychee link checher
+https://roidelapluie.be
+https://github.com/purpleidea/mgmt/commit
```
Why do we want to exclude this URL?

purpleidea reviewed (commented):
Patch looks good. One quick Q. But otherwise should we merge?

purpleidea commented:
I got a bit bit busy what with starting a new job, and so I completely forgot to merge this. Please accept my apologies, and if you'd like to privately email me your address, I will send you a few mgmt stickers as a thankyou/apology.

Patch has been rebased and merged in: 201cf091d53e595966c0895f6dacd4da49ee7f4a

\o/
